### PR TITLE
Fixes#112: add sortable attributes

### LINKED ISSUE #112: Add sortable attributes option
**Description**
In keeping with the common theme of the other MeiliSearch libraries, it would be nice to also have the sortableAttributes option within this library as well.

**Basic example**
```ruby
  meilisearch if: :published_at? do
    attribute :category_list, :name, :description, :number
    sortable_attributes [:number]
  end
```

**Other**
Any other things you want to add.

### PR DESCRIPTION
**Pull request by cjilbert504**
# Pull Request

## What does this PR do?
Fixes #112 

## PR checklist
Please check if your PR fulfills the following requirements:
- [X] Does this PR fix an existing issue?
- [X] Have you read the contributing guidelines?
- [X] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to MeiliSearch!

### CODE REVIEW

**Comment by brunoocasali** [comment]
> @brunoocasali - I see the linter failed on this, I think there are several places we could remove an empty line to get this passing if it's a rule that you care about enforcing. Let me know what you would like me to do here to address this issue. Thanks!

Yeah, you could re-generate the rubocop_todo.yml using `bundle exec rubocop --auto-gen-config`. 
This file needs a serious refactor and I don’t have the time to plan and do that yet 😅 !

File: lib/meilisearch-rails.rb [hunk 1]
```diff
@@ -50,6 +50,7 @@ class IndexSettings
       OPTIONS = %i[
         searchableAttributes
         filterableAttributes
+        sortableAttributes
```
Comment (brunoocasali) [inline lib/meilisearch-rails.rb:53]: Hey @cjilbert504 thanks a lot for your contribution! ❤️ 

Could you provide some simple testing use-case for this option? 🐈

**Review by brunoocasali — COMMENTED** [review]
Hey @cjilbert504 what do you think about adding some tests to this new behavior?
Just to make sure everything is ok!

Let me know if you need anything 😃

File: lib/meilisearch-rails.rb (hunk 1 above)
Comment (cjilbert504) [reply lib/meilisearch-rails.rb:53]: Yep, I can do that, thanks for getting back to me.

File: spec/integration_spec.rb [hunk 2]
```diff
@@ -829,6 +830,14 @@ class SerializedDocument < ActiveRecord::Base
     expect(facets.size).to eq(1)
     expect(facets).to include(black)
   end
+
+  it 'searches with sorting' do
+    blue = Color.create!(name: 'blue', short_name: 'blu', hex: 0x0000FF)
+    black = Color.create!(name: 'black', short_name: 'bla', hex: 0x000000)
+    green = Color.create!(name: 'green', short_name: 'gre', hex: 0x00FF00)
+    facets = Color.search('*', { sort: ['name:asc'] })
+    expect(facets).to eq([black, blue, green])
```
Comment (brunoocasali) [inline spec/integration_spec.rb:839]: I always suggest this improvement, looks like a silly thing, but it helps a lot when we are reading the test!

ref: http://blog.plataformatec.com.br/2014/04/improve-your-test-readability-using-the-xunit-structure/

```suggestion

    facets = Color.search('*', { sort: ['name:asc'] })

    expect(facets).to eq([black, blue, green])
```

**Review by brunoocasali — CHANGES_REQUESTED** [review]
@cjilbert504 I left a comment for you regarding the test you sent!

I'll merge after that, thank you a lot for the tests!

File: spec/integration_spec.rb (hunk 2 above)
Comment (cjilbert504) [reply spec/integration_spec.rb]: Thanks for linking that article! I can definitely see the readability improvements by leveraging this approach.

**Comment by cjilbert504** [comment]
@brunoocasali - Thank you for the feedback and for working with me to get this merged in!

**Comment by brunoocasali** [comment]
Unfortunately @cjilbert504 your tests failed because we are deeply connected with the order of the tests today. This breaking action reminded me to raise a new issue #115.

To help you fix the problem, I suggest you add at the top of your spec this line `Color.delete_all`.

```rb
it 'searches with sorting' do
    Color.delete_all

    blue = Color.create!(name: 'blue', short_name: 'blu', hex: 0x0000FF)
    black = Color.create!(name: 'black', short_name: 'bla', hex: 0x000000)
    green = Color.create!(name: 'green', short_name: 'gre', hex: 0x00FF00)

    facets = Color.search('*', { sort: ['name:asc'] })

    expect(facets).to eq([black, blue, green])
  end
```

Thank will make your tests pass. We will address this randomization issue someday 😅.

**Comment by cjilbert504** [comment]
No problem at all, @brunoocasali. Just updated with your suggestion, thanks again for the help here!

**Review by brunoocasali — APPROVED** [review]
Thank you for your awesome work and patience @cjilbert504 😅 

LGTM! 🎉 

Thanks a lot for contributing with Meilisearch ❤️

**Comment by brunoocasali** [comment]
bors merge

**Pull request merged**